packages/squiggle-lang/src/rescript/Utility/XYShape.res

```diff
@@ -60,6 +60,41 @@ module T = {
   let fromZippedArray = (pairs: array<(float, float)>): t => pairs |> Belt.Array.unzip |> fromArray
   let equallyDividedXs = (t: t, newLength) => E.A.Floats.range(minX(t), maxX(t), newLength)
   let toJs = (t: t) => {"xs": t.xs, "ys": t.ys}
+
+  module Validator = {
+    let fnName = "XYShape validate"
+    let property = (propertyName: string): Errors.property => {
+      fnName: fnName,
+      propertyName: propertyName,
+    }
+    let notSortedError = (p: string): Errors.error => NotSorted(property(p))
+    let notFiniteError = (p, exampleValue): Errors.error => NotFinite(property(p), exampleValue)
+    let isEmptyError = (propertyName): Errors.error => IsEmpty(property(propertyName))
+    let differentLengthsError = (t): Errors.error => DifferentLengths({
+      fnName: fnName,
+      p1Name: "Xs",
+      p2Name: "Ys",
+      p1Length: E.A.length(xs(t)),
+      p2Length: E.A.length(ys(t)),
+    })
+
+    let areXsSorted = (t: t) => E.A.Floats.isSorted(xs(t))
+    let areXsEmpty = (t: t) => E.A.length(xs(t)) == 0
+    let getNonFiniteXs = (t: t) => t->xs->E.A.Floats.getNonFinite
+    let getNonFiniteYs = (t: t) => t->ys->E.A.Floats.getNonFinite
+
+    let validate = (t: t) => {
+      let xsNotSorted = areXsSorted(t) ? None : Some(notSortedError("Xs"))
+      let xsEmpty = areXsEmpty(t) ? Some(isEmptyError("Xs")) : None
+      let differentLengths =
```

Comment:
We wouldn't need this check if we did the #328, but we'll keep it for now

Reply:
One of the intentions is to eventually make a smart constructur for XYShape, and for this validator to be used for that constructor.